syntax/puppet.vim

```diff
@@ -1,167 +1,433 @@
-" puppet syntax file
-" Filename:     puppet.vim
-" Language:     puppet configuration file
-" Maintainer:   Luke Kanies <luke@madstop.com>
-" URL:
-" Last Change:
-" Version:
```

Comment:
I didn't include them because they had no content. Some comments regarding them:
* URL: Sure, it could point to this repo.
* Last Change: Can be determined by looking at the URL. If present, it should probably be updated automatically in each commit or it will diverge. What scheme should be used for this?
* Version: What versioning scheme do you want? A SHA? A Semver? Regardless, why not use a version file that denotes the version currently in the repository. It's not unlikely that future improvements span more than this particular file.
Filename: What's the point in including a file name in a named file?